tokenizer.h

```diff
@@ -0,0 +1,93 @@
+/*
+Defines the GPT-2 Tokenizer.
+Only supports decoding, i.e.: tokens (integers) -> strings
```

Comment:
Only supports **decoding**, i.e.: tokens (integers) -> strings
This is all we need for unconditional generation.
If we wanted to later prompt the model, we'd have to add **decoding**  :)

Reply:
hehe oops